src/controller/AuthController.ts

```diff
@@ -0,0 +1,38 @@
+import { Request, Response } from 'express';
+import { CustomError } from '../interfaces/CustomError';
+import { AuthService } from '../services/AuthService';
+
+/**
+ * Função utilitária para lançar um erro personalizado.
+ *
+ * @param message - A mensagem do erro.
+ * @param status - O código de status HTTP associado ao erro (opcional, padrão 400).
+ * @throws {CustomError} - Lança um erro personalizado.
+ */
+export const throwCustomError = (
```

Comment:
Aqui você está exportando o erro, porém ele só é utilizado internamente no arquivo.
Se for utilizar externamente, recomendo criar uma pasta separada para deixar os tratamentos de erros desse tipo ou até mesmo um arquivo de util para ele.